### ISSUE TITLE
Suggestion For New Lint: Check Lints Section in `Cargo.toml` for Typos

### ISSUE THREAD

**Issue by mre** [rust-lang/rust-clippy#14541]
### What it does

Currently, misspelled or non-existent lint names aren't detected when configured in `Cargo.toml`, causing a silent failure.

```toml
[package]
name = "foo"
edition = "2024"

[lints.clippy]
doesnt_exist = "deny"  # Typo isn't detected or reported
```

The project builds without any warnings, potentially hiding configuration mistakes.
In contrast, the same non-existent lint configured in a Rust file produces a helpful warning:

```rust
// main.rs
#![deny(doesnt_exist)]
```

```shell
❯❯❯ cargo check
    Checking foo v0.0.0
warning: unknown lint: `doesnt_exist`
 --> src/main.rs:3:9
  |
3 | #![deny(doesnt_exist)]
  |         ^^^^^^^^^^^^
  |
  = note: `#[warn(unknown_lints)]` on by default

warning: `foo` (bin "foo") generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.10s
```

Could we add a lint (or another kind of validation mechanism) to detect and warn about unknown lint names in `Cargo.toml`?

### Advantage

This would be especially valuable since defining lints in `Cargo.toml` is great for workspace-wide configurations, but currently carries the risk of silently ignoring typos.

### Drawbacks

None that I'm aware of.

### Example

The output when using the above `Cargo.toml` could be:

```shell
❯❯❯ cargo check
    Checking foo v0.0.0
warning: unknown lint: `doesnt_exist`
 --> Cargo.toml:6:0
  |
6 | doesnt_exist = "deny"
  |  ^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unknown_lints)]` on by default

warning: `foo` (bin "foo") generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.10s
```

Or, alternatively, it could print a warning when running `cargo clippy`.

**Comment by Alexendoo** [comment]
You should see a warning when you run `cargo clippy` like

```
warning[E0602]: unknown lint: `clippy::doesnt_exist`
  |
  = note: requested on the command line with `-D clippy::doesnt_exist`
  = note: `#[warn(unknown_lints)]` on by default
```

**Comment by mre** [comment]
Ah, you're right. Sorry about that. Must have overlooked it while doing some refactoring. All good then. 😅
